execution_engine/src/shared/storage_costs.rs

```diff
@@ -0,0 +1,67 @@
+use bytesrepr::{FromBytes, ToBytes};
+use datasize::DataSize;
+use rand::{distributions::Standard, prelude::*, Rng};
+use serde::{Deserialize, Serialize};
+
+use casper_types::bytesrepr;
+
+#[derive(Copy, Clone, PartialEq, Eq, Serialize, Deserialize, Debug, DataSize)]
+pub struct StorageCosts {
+    /// Gas charged per byte stored in the global state.
+    pub gas_per_byte: u32,
```

Comment:
Why do we need to make this a public field?

Reply:
I'd prefer if that's possible to keep config attributes like this (incl. host functions, opcode costs, etc) to indicate they're dynamic and can grow or shrink over time and also to avoid creating big `fn new` methods, or introducing builder patterns etc.

I also made nested structures of `WasmCosts` private and added accessors for these to also indicate they're integral part of the API

Reply:
:+1:

Reply:
:+1: